src/main/java/com/datapipe/jenkins/vault/log/MaskingConsoleLogFilter.java

```diff
@@ -17,35 +18,30 @@ public class MaskingConsoleLogFilter extends ConsoleLogFilter
     private static final long serialVersionUID = 1L;
 
     private final String charsetName;
-    private final List<String> valuesToMask;
+    private final Pattern pattern;
+
 
 
     public MaskingConsoleLogFilter(final String charsetName,
         List<String> valuesToMask) {
         this.charsetName = charsetName;
-        this.valuesToMask = valuesToMask;
+
+        // Filter out null values
+        List<String> values = valuesToMask.stream().filter(Objects::nonNull).collect(Collectors.toList());
+
+        if (!valuesToMask.isEmpty()) {
+            this.pattern = SecretPatterns.getAggregateSecretPattern(values);
+        } else {
+            this.pattern = null;
+        }
     }
 
     @Override
     public OutputStream decorateLogger(Run run,
         final OutputStream logger) throws IOException, InterruptedException {
         return new SecretPatterns.MaskingOutputStream(logger,
-            () -> {
-                // Only return a non-null pattern once there are secrets to mask. When a non-null
-                // pattern is returned it is cached and not supplied again. In cases like
-                // VaultBuildWrapper the secrets are added to the "valuesToMasker" list AFTER
-                // construction AND AFTER the decorateLogger method is initially called, therefore
-                // the Pattern should only be returned once the secrets have been made available.
```

Comment:
This could not have worked anyway when `decorateLogger` was run on the agent side, since the mutation would be to the `List` stored on the controller side, but this has already been serialized and copied to the agent. So the plugin may have been silently failing to mask some secrets.

Reply:
…or not, I guess, since `VaultBuildWrapper` would only be called in this way from a freestyle project, which would never call `decorateLogger` from the agent side.